gh-pages/assets/javascript/map.js

```diff
@@ -0,0 +1,204 @@
+
+import "./vendor/leaflet.js"
+
+const L = window["L"]
+
+const COLORS = ["#FFEDA0", "#FED976", "#FEB24C", "#FD8D3C", "#FC4E2A", "#E31A1C", "#BD0026", "#800026"]
+
+/**
+ * Load a GeoJson of the french departments
+ * @returns {Promise} a promise of a GeoJson
+ */
+async function loadFrenchGeojson() {
+    const response = await fetch("./assets/geojson/french_departments.json")
+    return await response.json()
+}
+
+/**
+ * Create an info element to display on the top right of the map
+ * @param {string} title Title of the info element, undefined to use none
+ * @returns The created info element. Use ``info.addTo(map)``.
+ */
+function createInfoElement(title=undefined) {
+    let info = L.control()
+    info.onAdd = function () {
+        this._div = L.DomUtil.create("div", "info")
+        this.update()
+        return this._div
+    }
+    info.update = function (props) {
+        let html = ""
+        if (title) {
+            html += "<h4>" + title + "</h4>"
+        }
+        if (props) {
+            html += "<b>Department</b>: " + props.nom + " (" + props.code + ")" + "<br />"
+            for (let title in props.info) {
+                html += "<b>" + title + "</b>: " + props.info[title] + "<br />"
+            }
+        } else {
+            html += "Hover over a state"
+        }
+        this._div.innerHTML = html
+    }
+    return info
+}
+
+
+/**
+ * French Map object. Usable to display statistics on french departments.
+ */
+export class FrenchMap {
+    constructor() {
+        this.map = L.map("map").setView([46.6309, 2.4527], 5)
+        L.tileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
+            maxZoom: 10,
+            minZoom: 5,
+            attribution: "© OpenStreetMap"
```

Comment:
- [ ] Use constants for center coordinates and zoom limits ?
- [ ] Use a link for attribution

Reply:
- [x] [Extract map-related constants](https://github.com/Doreapp/prix-carburants/pull/36/commits/aa74259b0fdd44c6609d4b3d08c1325f5d68a34a)
- [x] [Improve OSM attribution](https://github.com/Doreapp/prix-carburants/pull/36/commits/b6f57e6bc0380a4e6b67e8c2a4e6d3c8762d63e5)